### PR TITLE
Original morse code converter

### PR DESCRIPTION
**Pull request by chayahaor**

### CODE REVIEW

**Comment by andrewoid** [comment]
Pretty sure none of this code actually works.

File: src/main/java/MorseCodeConverter.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class MorseCodeConverter {
+    static Map<String, String> morseCodeMap = new HashMap<>();
```
Comment (andrewoid) [inline src/main/java/MorseCodeConverter.java]: As I mentioned in class more than once, static variables are not allowed.

File: src/main/java/MorseCodeFrame.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+
+public class MorseCodeFrame extends JFrame {
```
Comment (andrewoid) [inline src/main/java/MorseCodeFrame.java:5]: You should be fixing MorseCodeConverter before working on anything else.

File: src/test/java/MorseCodeConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class MorseCodeConverterTest {
+
+    @Test
+    public void testFillMorse() {
```
Comment (andrewoid) [inline src/test/java/MorseCodeConverterTest.java]: You don't need to start your test method names with the word test. This is an old convention that isn't used anymore.

**Comment by andrewoid** [comment]
Add in the YAML file so that GitHub runs your tests.

**Review by andrewoid — CHANGES_REQUESTED** [review]

File: src/main/java/MorseCodeConverter.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        mapToMorse.put(".....", "5");
+        mapToMorse.put("-....", "6");
+        mapToMorse.put("--...", "7");
+        mapToMorse.put("---..", "8");
+        mapToMorse.put("----.", "9");
+    }
+
+    String toMorse(String english) {
+        Map<String, String> mapToMorse = new HashMap<>();
+        fillMapToMorse(mapToMorse);
```
Comment (andrewoid) [inline src/main/java/MorseCodeConverter.java]: Why do you need to do this every time you call this method?

File: src/test/java/MorseCodeConverterTest.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+class MorseCodeConverterTest {
+
+    @Test
+    public void toEnglish() {
+        //given
+        MorseCodeConverter convert = new MorseCodeConverter();
+        //when
+        String morse = ".... . .-.. .-.. ---";
+        //then
+        assertEquals("h e l l o ", convert.toEnglish(morse));
```
Comment (andrewoid) [inline src/test/java/MorseCodeConverterTest.java]: This String shouldn't end with a space.

File: src/test/java/MorsePresenterTest.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    MorseCodeFrame view = mock(MorseCodeFrame.class);
+    MorseCodeConverter model = mock(MorseCodeConverter.class);
+    MorsePresenter presenter = new MorsePresenter(view, model);
+
+    @Test
+    public void convertEnglish() {
+        //given
+
+        //when
+        String toConvert = ".... . .-.. .-.. ---";
```
Comment (andrewoid) [inline src/test/java/MorsePresenterTest.java]: This belongs in the given

File: src/test/java/MorsePresenterTest.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        presenter.convertToEnglish(toConvert);
+        //then
+
+        verify(model).toEnglish(toConvert);
+        verify(view).setTaEnglishInput(toConvert);
```
Comment (andrewoid) [inline src/test/java/MorsePresenterTest.java]: Does setTaEnglishInput get called with toConvert?

Comment (chayahaor) [reply src/test/java/MorsePresenterTest.java]: When calling setTaEnglishInput it takes in a string toConvert that becomes the text in the text area.

Comment (andrewoid) [reply src/test/java/MorsePresenterTest.java]: No. It doesn't take the toConvert string.

Comment (chayahaor) [reply src/test/java/MorsePresenterTest.java]: I'm confused. I set the method to take a string that is in morse code to be converted to English.
Are you saying the class method does not take it? or that the mockito method does not take it?

Comment (andrewoid) [reply src/test/java/MorsePresenterTest.java]: In your test, "model" is a mock.
When you call a method on a mock, what does it return?

File: src/main/java/MorseCodeConverter.java (hunk 4 above)
Comment (andrewoid) [reply src/main/java/MorseCodeConverter.java]: This is a waste. You need to fix this before I'm going to approve this PR.

File: src/main/java/MorseCodeConverter.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class MorseCodeConverter {
+    Map<String, String> mapToMorse = new HashMap<>();
```
Comment (andrewoid) [inline src/main/java/MorseCodeConverter.java]: Member variables should be private

File: src/main/java/MorseCodeConverter.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+    Map<String, String> mapToEnglish = new HashMap<>();
+
+    public MorseCodeConverter(){
+        fillMapToMorse(mapToMorse);
+        fillMapToEnglish(mapToEnglish);
```
Comment (andrewoid) [inline src/main/java/MorseCodeConverter.java]: Because these are now member variables, they don't need to be passed into the method.

**Review by andrewoid — APPROVED** [review]

File: src/main/java/MorseCodeConverter.java [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+public class MorseCodeConverter {
+    private Map<String, String> mapToMorse = new HashMap<>();
+    private Map<String, String> mapToEnglish = new HashMap<>();
+
+    public MorseCodeConverter() {
+        fillMapToMorse();
+        fillMapToEnglish();
+    }
+
+    void fillMapToMorse() {
```
Comment (andrewoid) [inline src/main/java/MorseCodeConverter.java]: and these should be private

**Review by andrewoid — APPROVED** [review]

**Review by andrewoid — APPROVED** [review]

**Pull request merged**